contract_payment_mode/models/contract.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    payment_mode_id = fields.Many2one(
+        'payment.mode',
+        string='Payment Mode',
+        domain="[('type', '=', 'sale')]")
+
+    @api.multi
+    def on_change_partner_id(self, partner_id, name):
+        res = super(AccountAnalyticAccount, self).on_change_partner_id(
```

Comment:
I wrote wrong: it's `self.ensure_one()` instead.
It's a safeguard since you assume in the rest of the code that you're dealing with a singleton.
But maybe it redundant for on_change. What do you think @pedrobaeza ?